service/src/main/java/bio/terra/cbas/controllers/MethodsApiController.java

```diff
@@ -84,19 +87,49 @@ public ResponseEntity<PostMethodResponse> postMethod(PostMethodRequest postMetho
 
     String methodSource = postMethodRequest.getMethodSource().toString();
 
+    // convert method url to raw GitHub url and use that to call Cromwell's /describe endpoint
+    String rawMethodUrl;
+    try {
+      rawMethodUrl =
+          MethodUtil.convertToRawGithubUrl(
+              postMethodRequest.getMethodUrl(),
+              postMethodRequest.getMethodSource(),
+              postMethodRequest.getMethodVersion(),
+              dockstoreService);
+
+      // this could happen if there was no url or empty url received in the Dockstore workflow's
+      // descriptor response
+      if (rawMethodUrl == null || rawMethodUrl.isEmpty()) {
+        String errorMsg =
+            "Error while importing Dockstore workflow. No workflow url found at specified path.";
+        log.error(errorMsg);
+        return new ResponseEntity<>(
+            new PostMethodResponse().error(errorMsg), HttpStatus.BAD_REQUEST);
+      }
+    } catch (URISyntaxException | MalformedURLException e) {
+      String errorMsg =
+          "Bad user request. Method url has invalid value. Error: %s".formatted(e.getMessage());
+      log.error(errorMsg, e);
+      return new ResponseEntity<>(new PostMethodResponse().error(errorMsg), HttpStatus.BAD_REQUEST);
+    } catch (bio.terra.dockstore.client.ApiException e) {
+      String errorMsg =
+          "Error while importing Dockstore workflow. Error: %s".formatted(e.getMessage());
+      log.error(errorMsg, e);
+      return new ResponseEntity<>(new PostMethodResponse().error(errorMsg), HttpStatus.BAD_REQUEST);
+    }
+
     // call Cromwell's /describe endpoint to get description of the workflow along with inputs and
     // outputs
```

Comment:
TOL: This method is getting a little long and unwieldy. I'm finding it a little hard to keep track of everything that it's doing, and I think this kind of comment is also maybe a bit of a sign that we could break it down into a couple of sub-functions, to keep the top-level coordination function shorter and easier to read?

Reply:
I had only touched the code inside this block to replace the method.url with the rawUrl. But I can try and refactor this method to make it more readable 👍

Reply:
I have refactored this method a bit. I think because of various exceptions handlings it is a little verbose but maybe its little more readable now?